### ISSUE TITLE
A trailing / in a URL redirect will result in an incorrect url redirect

### ISSUE THREAD

**Issue by myott** [ratpack/ratpack#813]
A trailing `/` in a URL redirect will result in an incorrect url redirect. 

- This was first noticed on the Ratpack site by navigating to `ratpack.io/manual/` which 301s to `ratpack.io/manual/manual/current`
- I can reproduce this via the following test (I created) in `ratpack.http.client`

```java
  def "Trailing / should be handled"() {
    given:
    otherApp {
      get("foo") {
        redirect(301, "foo/bar")
      }
      get("foo/bar") { render "baz" }
    }

    when:
    handlers {
      get { HttpClient httpClient ->
        httpClient.get(otherAppUrl("foo/")) {
        } then { ReceivedResponse response ->
          render response.body.text
        }
      }
    }

    then:
    text == "baz"
  }
```

This will result in:
```
ERROR r.e.i.DefaultDevelopmentErrorHandler [ratpack-compute-7-3] 404 client error for request to /foo/foo/bar 
```

Is this expected behavior?

**Comment by danhyun** [comment]
As far as this being intentional behavior, it seems as though we should be able to handle this.

For the time being there is a way to get this assertion to pass:

```
def "Trailing / should be handled"() {
    given:
    otherApp {
      get("foo/:part?") { render 'baz' }
    }

    when:
    handlers {
      get { HttpClient httpClient ->
        httpClient.get(otherAppUrl("foo/")).then { ReceivedResponse response ->
            render response.body.text
        }
      }
    }

    then:
    text == "baz"
  }
```

The Path Binding javadoc demonstrates the ability to declare an optional path token http://ratpack.io/manual/snapshot/api/ratpack/handling/Chain.html#path-binding which will match the incoming request.

**Comment by danhyun** [comment]
Actually this scenario can be simplified to demonstrate the issue at hand:

```
class SimpleSpec extends RatpackGroovyDslSpec {
  def "Trailing / should be handled"() {
    given:
    handlers {
      get("foo/:part?") { render 'baz' }
    }

    expect:
    getText('foo') == 'baz'
    getText('foo/') == 'baz'
    getText('foo/baz') == 'baz'
  }
}
```

**Comment by danhyun** [comment]
After carefully rereading the issue I realize this is about redirects ending up at a funky path.

It seems that providing an absolute uri will fix this:

```
class SimpleSpec extends RatpackGroovyDslSpec {
  def "Trailing / should be handled"() {
    given:
    handlers {
      get("foo") { redirect 301, '/foo/bar' }
      get("foo/bar") { render 'baz' }
    }

    expect:
    getText('foo') == 'baz'
    getText('foo/') == 'baz'
    getText('foo/bar') == 'baz'
  }
}
```

**Comment by myott** [comment]
@danhyun Is this still worth fixing? Seems it is still an "issue" in 1.2.0.

**Comment by danhyun** [comment]
I would add unit tests of demonstrating expected behavior.

This may be a situation that calls for improved documentation but I know that currently navigating to ratpack.io/manual/ will redirect to ratpack.io/manual/manual/current when it should be ratpack.io/manual/current.

So in this case we probably want updated javadoc and updated redirect in `ratpack-site`, are you up for that?

**Comment by myott** [comment]
You've got it.

**Comment by myott** [comment]
@danhyun I opened a PR, https://github.com/ratpack/ratpack/pull/939

Note that I did not change any documentation as I believe this is quite intuitive. The only confusion was whether or not this specific case (the manual redirect) was intended. This seems to boil down to simply a bug in `ratpack-site`.

**Comment by danhyun** [comment]
Fixed via #939
